lib/retry.js

```diff
@@ -4,37 +4,67 @@ const sleep = msec => new Promise(resolve => {
   setTimeout(resolve, msec);
 });
 
-const promisify = fn => (...args) => (
-  new Promise((resolve, reject) => (
-    fn(...args, (err, data) => (
-      err ? reject(err) : resolve(data)
-    ))
-  ))
-);
+const promisify = (fn, args) => new Promise((resolve, reject) => {
+  if (!args) args = [];
+  args.push((err, data) => (err ? reject(err) : resolve(data)));
+  fn(...args);
+});
 
-const retry = async (times, fn, ...args) => {
-  let tries = 0;
-  let done = false;
-  let res = null;
-  const fnR = fn;
-  if (args[args.length - 1] instanceof Function) {
-    fnR = promisify(fn);
-    args.pop();
+const deepClone = obj => {
+  if (obj === null || typeof obj !== 'object') {
+    return obj;
+  } else if (Array.isArray(obj)) {
+    const result = [];
+    for (const data of obj) {
+      result.push(deepClone(data));
+    }
+    return result;
+  } else {
+    const result = {};
+    const keys = Object.keys(obj);
+    for (const key of keys) {
```

Comment:
Maybe use `for(key in obj)` instead?

Reply:
I've been thinking about it. It seems to me that the code will be more beautiful if I use one kind of loops in this function. What do you think?